debezium-core/src/main/java/io/debezium/processors/PostProcessorRegistryServiceProvider.java

```diff
@@ -31,17 +33,32 @@ public class PostProcessorRegistryServiceProvider implements ServiceProvider<Pos
 
     private static final String POST_PROCESSOR_MISS_CONFIGURATION_ERROR_MESSAGE = "Post processor '%s' is missing '%s.type' and/or '%s.<option>' configurations";
     private final String TYPE_SUFFIX = ".type";
+    private final ServiceLoader<PostProcessorFactory> postProcessorFactory = ServiceLoader.load(PostProcessorFactory.class);
```

Comment:
I would prefer a slightly different architectural & strategic change here.

Rather than using `ServiceLoader` directly & side-step the `ServiceRegistry`, could we instead create a dedicated dependency injection service & matching provider, of which there would be a Quarkus/Arc implementation, and this would lay the framework for future DI frameworks like Spring.

Then here we ask the `ServiceRegistry` for this DI service, it would return us the Quarkus/Arc implementation behind that common DI service interface, and we can then ask to get a list of all beans with a given interface.

This means we pay the cost for this hook once and we can use it for PostProcessors or anything in the the future that we want to hook in.  What do you think @kmos ?

Reply:
That's not a bad idea, and it gives us the chance to take a birds eye view of the picture to make sure that we design the Quarkus/Arc/DI stuff well that fits all our needs and leaves room for alternatives.